src/pages/index.js

```diff
@@ -6,87 +6,230 @@ import {
+pictureFormValidator.enableValidation();
 
 // FUNCTIONS //
 function handleImageClick(card) {
   cardPreviewClass.open({ name: card._name, link: card._link });
 }
 
-function handleCardAddSubmit(data) {
-  renderCard({ name: data.title, link: data.link });
-  cardAddFormClass.close();
-  cardAddForm.reset();
+function handleCardAddSubmit(inputData) {
+  cardAddFormClass.isLoading(true);
+  api
+    .addCard(inputData)
+    .then((data) => {
+      const { _id, name, link, isLiked } = data;
+      const newCardData = { _id, name, link, isLiked };
+      cardSection.addItem(createCard(newCardData));
+      addFormValidator.disableBtn();
+      cardAddForm.reset();
+      cardAddFormClass.close();
+    })
+    .catch((err) => {
+      console.error(`Error: failed to add card. ${err}`);
+    })
+    .finally(() => {
+      cardAddFormClass.isLoading(false);
+    });
 }
 
 function handleProfileFormSubmit(data) {
-  profileInfoClass.setUserInfo(data);
-  profileFormClass.close();
+  console.log(data);
+  profileFormClass.isLoading(true);
+  api
+    .setUserInfo(data)
+    .then(() => {
+      profileInfoClass.setUserInfo({ name: data.name, about: data.about });
+      editFormValidator.disableBtn();
+      profileFormClass.close();
+    })
+    .catch((err) => {
+      console.error(`Error handling profile form submit. ${err}`);
+    })
+    .finally(() => {
+      profileFormClass.isLoading(false);
+    });
 }
 
-function renderCard(card) {
-  const cardElementClass = new Card({
-    name: card.name,
-    link: card.link,
-    cardSelector: "#card-template",
-    handleImageClick: handleImageClick,
+function handlePictureSubmit(avatarUrl) {
+  pictureFormClass.isLoading(true);
+  api
+    .changeAvatar(avatarUrl)
+    .then(() => {
+      profileInfoClass.setUserInfo({ avatar: avatarUrl.picture });
+      pictureFormValidator.disableBtn();
+      pictureFormClass.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      pictureFormClass.isLoading(false);
+    });
+}
+
+function handleLikeBtn(card) {
+  api
+    .setCardLiked(card.id, card.getLikedStatus())
+    .then((updatedCard) => {
+      card.handleIsLiked(updatedCard.isLiked);
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+}
+
+function handleDeleteBtn(card) {
+  confirmFormClass.open();
+  confirmFormClass.handleConfirm(() => {
+    confirmFormClass.viewLoading(true);
+    api
+      .removeCard(card.id)
+      .then(() => {
+        card.removeCard();
+        confirmFormClass.viewLoading(false);
+        confirmFormClass.close();
+      })
+      .catch((err) => {
+        console.error(`Failed to delete card. ${err}`);
+        confirmFormClass.viewLoading(false);
+      });
   });
-  cardSectionClass.addItem(cardElementClass.getView());
 }
 
 // LISTENERS //
+pictureEditBtn.addEventListener("click", () => {
+  pictureFormValidator.resetValidation();
+  pictureLinkInput.value = pictureEl.src;
```

Comment:
Please, delete it

You don't need to insert the link because you can't edit it


And `pictureEl.src;` shouldn't be used. You have `profileInfoClass.getUserInfo();` for getting this data

Reply:
I am also confused here, too.  I thought this was necessary to open the avatar modal? 

Or is this saying I should change the constants name to something more fitting like avatarChangeBtn?

I did update the pictureEl.src to profileInfoClass.getUserInfo().avatar;